app/script/main/app.js

```diff
@@ -559,48 +558,19 @@ z.main.App = class App {
    * Check that this is the single instance tab of the app.
    * @returns {Promise} Resolves when page is the first tab
    */
-  _checkSingleInstanceOnInit() {
-    if (!z.util.Environment.electron) {
-      return this._setSingleInstanceCookie();
-    }
-
-    return Promise.resolve();
-  }
-
-  _checkSingleInstanceOnInterval() {
-    const singleInstanceCookie = Cookies.getJSON(App.CONFIG.TABS_CHECK.COOKIE_NAME);
-
-    const shouldBlockTab = !singleInstanceCookie || singleInstanceCookie.appInstanceId !== this.instanceId;
-    if (shouldBlockTab) {
-      return this._redirectToLogin(z.auth.SIGN_OUT_REASON.MULTIPLE_TABS);
+  _registerSingleInstance() {
+    if (this.singleInstanceHandler.registerInstance(this.instanceId)) {
+      this.singleInstanceHandler.addExtraInstanceStartedListener(this._onExtraInstanceStarted);
```

Comment:
We can further simplify the API if the register instance takes the listener as param right away. No need for an additional step. Currently we don't even need to support multiple listeners. It is overkill or do you see that need for us?

Reply:
No, your totally right. I first implemented this thinking that the login view would also need to listen to this. 
This is overkill indeed.

I think I can come up with something better. Give me a few minutes :D